Test/DurableTask.AzureStorage.Tests/AzureTableTrackingStoreTest.cs

```diff
@@ -0,0 +1,245 @@
+    using System.Threading.Tasks;
+    using DurableTask.AzureStorage.Monitoring;
+    using DurableTask.AzureStorage.Tracking;
+    using DurableTask.Core;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.WindowsAzure.Storage.Table;
+    using Moq;
+    using Newtonsoft.Json;
+
+    [TestClass]
+    public class AzureTableTrackingStoreTest
+    {
+        [TestMethod]
+        public async Task QueryStatus_WithContinuationToken_NoInputToken()
+        {
+            var fixture = new QueryFixture();
+            fixture.SetUpQueryStateWithPagerWithoutInputToken();
+
+            var inputState = new List<OrchestrationStatus>();
+            inputState.Add(OrchestrationStatus.Running);
+            inputState.Add(OrchestrationStatus.Completed);
+            inputState.Add(OrchestrationStatus.Failed);
+
+            var result = await fixture.TrakingStore.GetStateAsync(fixture.ExpectedCreatedDateFrom, fixture.ExpectedCreatedDateTo, inputState, 3, fixture.InputToken);
+
+            Assert.IsNull(fixture.ActualPassedTokenObject);
+
+            Assert.AreEqual(fixture.ExpectedResult.ContinuationToken, result.ContinuationToken);
+            Assert.AreEqual(fixture.ExpectedResult.OrchestrationState.Count(), result.OrchestrationState.Count());
+            Assert.AreEqual(fixture.ExpectedResult.OrchestrationState.FirstOrDefault().Name, result.OrchestrationState.FirstOrDefault().Name);
+            fixture.VerifyQueryStateWithPager();
+        }
+
+        [TestMethod]
+        public async Task QueryStatus_WithContinuationToken_InputToken()
+        {
+            var fixture = new QueryFixture();
+            var inputToken = new TableContinuationToken()
+            {
+                NextPartitionKey = "qux",
+                NextRowKey = "quux",
+                NextTableName = "corge",
+            };
+
+            fixture.SetupQueryStateWithPagerWithInputToken(inputToken);
+
+            var inputState = new List<OrchestrationStatus>();
+            inputState.Add(OrchestrationStatus.Running);
+            inputState.Add(OrchestrationStatus.Completed);
+            inputState.Add(OrchestrationStatus.Failed);
+
+            var result = await fixture.TrakingStore.GetStateAsync(fixture.ExpectedCreatedDateFrom, fixture.ExpectedCreatedDateTo, inputState, 3, fixture.InputToken);
+
+            Assert.AreEqual(inputToken.NextPartitionKey, fixture.ActualPassedTokenObject.NextPartitionKey);
+        
+            Assert.AreEqual(fixture.ExpectedResult.ContinuationToken, result.ContinuationToken);
+            Assert.AreEqual(fixture.ExpectedResult.OrchestrationState.Count(), result.OrchestrationState.Count());
+            Assert.AreEqual(fixture.ExpectedResult.OrchestrationState.FirstOrDefault().Name, result.OrchestrationState.FirstOrDefault().Name);
+            fixture.VerifyQueryStateWithPager();
+        }
+
+        private class QueryFixture
+        {
+            Mock<CloudTable> cloudTableMock;
```

Comment:
Make instance variables `readonly` whenever possible.
```suggestion
            private readonly Mock<CloudTable> cloudTableMock;
```

Reply:
Done by you